File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.swift
if let input: String = readLine() {
    let arrayOfInput = input.split(separator: " ").compactMap {Int($0)}
    // [Int] _ (Disabled): An Integer that represents length of mapOfN (range: 1 ≤ N ≤ 100,000)
    // [Int] M (immutuable): An Integer that represents count of test cases (range: 1 ≤ M ≤ 100,000)
    let (_, M): (Int, Int) = (arrayOfInput[0], arrayOfInput[1])

    // [Array<Int>] arrayOfN (immutable): Array of Integers that stores nature numbers to find prefix sum
    let arrayOfN: [Int] = (readLine() ?? "1").split(separator: " ").compactMap {Int($0)}
    
    // [Array<Int>] prefixSum (mutable): Array of Integers that stores prefix sum of mapOfN
    var prefixSum: [Int] = [0]
    
    // Update prefixSum
    for (idx, n) in arrayOfN.enumerated() {
        prefixSum.append( n + prefixSum[idx] )
    }
    
    for _ in 0..<M {
        // [Int] i, j (immutable): An integer representing the range to obtain the prefix sum (range: 1 ≤ i ≤ j ≤ N)
        let line: [Int] = (readLine() ?? "1 1").split(separator: " ").compactMap { Int($0) }
        let (i, j): (Int, Int) = (line[0], line[1])
    
        // Calculate sum of sections i to j
        print(prefixSum[j] - prefixSum[i-1])
    }
}

File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.py
import sys

input = sys.stdin.readline

# [int] N: An Integer that represents length of mapOfN
# [int] M: An Integer that represents count of test cases
N, M = map(int, input().split())

# [map(int)] mapOfN: Map of Integers that stores nature numbers to find prefix sum
mapOfN = map(int, input().split())

# [list(int)] prefixSum: List of Integers that stores prefix sum of mapOfN
prefixSum = [0]

# Update prefixSum
for idx, n in enumerate(mapOfN):
    prefixSum.append( n + prefixSum[idx] )

for _ in range(M):
    i, j = map(int, input().split())

    # Calculate sum of sections i to j
    print(prefixSum[j] - prefixSum[i-1])